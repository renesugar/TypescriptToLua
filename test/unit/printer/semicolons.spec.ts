import * as util from "../../util";

test.each(["const a = 1; const b = a;", "const a = 1; let b: number; b = a;", "{}", "function bar() {} bar();"])(
    "semicolon insertion (%p)",
    leadingStatement => {
        util.testFunction`
            let result = "";
            function foo() { result = "foo"; }
            ${leadingStatement}
            (undefined || foo)();
            return result;
        `
            .expectToMatchJsResult()
            .expectLuaToMatchSnapshot();
    }
);
