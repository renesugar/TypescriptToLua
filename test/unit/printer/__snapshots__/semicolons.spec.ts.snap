// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`semicolon insertion ("{}") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    local result = \\"\\"
    local function foo(self)
        result = \\"foo\\"
    end
    do
    end
    (nil or foo)(nil)
    return result
end
return ____exports"
`;

exports[`semicolon insertion ("const a = 1; const b = a;") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    local result = \\"\\"
    local function foo(self)
        result = \\"foo\\"
    end
    local a = 1
    local b = a;
    (nil or foo)(nil)
    return result
end
return ____exports"
`;

exports[`semicolon insertion ("const a = 1; let b: number; b = a;") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    local result = \\"\\"
    local function foo(self)
        result = \\"foo\\"
    end
    local a = 1
    local b
    b = a;
    (nil or foo)(nil)
    return result
end
return ____exports"
`;

exports[`semicolon insertion ("function bar() {} bar();") 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    local result = \\"\\"
    local function foo(self)
        result = \\"foo\\"
    end
    local function bar(self)
    end
    bar(nil);
    (nil or foo)(nil)
    return result
end
return ____exports"
`;
